### PR TITLE
pre-commit: Upload pytest junit artifact

### PR DESCRIPTION
**Pull request by dorschw**
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-9598
Demo [workflow run](https://github.com/demisto/content/actions/runs/7990514347/job/21819495927)

### CODE REVIEW

File: .pre-commit-config_template.yaml [hunk 1]
```diff
@@ -283,177 +283,175 @@ repos:
...
-    - types-ujson==5.8.0.1 ; python_version >= "3.8" and python_version < "3.11"
+    - types-tqdm==4.66.0.20240106 ; python_version >= "3.8" and python_version < "3.11"
+    - types-ujson==5.9.0.0 ; python_version >= "3.8" and python_version < "3.11"
     - types-urllib3==1.26.25.14 ; python_version >= "3.8" and python_version < "3.11"
-    - typing-extensions==4.8.0 ; python_version >= "3.8" and python_version < "3.11"
-    - tzdata==2023.3 ; python_version >= "3.8" and python_version < "3.11"
-    - tzlocal==4.3.1 ; python_version >= "3.8" and python_version < "3.11"
-    - ujson==5.8.0 ; python_version >= "3.8" and python_version < "3.11"
+    - typing-extensions==4.9.0 ; python_version >= "3.8" and python_version < "3.11"
+    - tzdata==2024.1 ; python_version >= "3.8" and python_version < "3.11" and platform_system == "Windows"
```
Comment (yuvalbenshalom) [inline .pre-commit-config_template.yaml:446]: please update me privately why this check of windows platform is relevant.

File: Utils/github_workflow_scripts/parse_junit_per_pack.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            f"Cannot parse pack name out of {value}, expected Packs.<pack_name>.more_parts"
+        )
+
+
+def parse_pack_name(class_name: str):
+    # parses Packs.IPINFO.Integrations.ipinfo_v2.ipinfo_v2_test into IPINFO
+    if (
+        (not class_name.startswith("Packs."))
+        or len(parts := class_name.split(".")) < 3
+        or not (parsed_pack_name := parts[1])
```
Comment (yuvalbenshalom) [inline Utils/github_workflow_scripts/parse_junit_per_pack.py:21]: this cannot have index error?

**Review by yuvalbenshalom — COMMENTED** [review]
looks good. see my questions.

File: Utils/github_workflow_scripts/parse_junit_per_pack.py (hunk 2 above)
Comment (dorschw) [reply Utils/github_workflow_scripts/parse_junit_per_pack.py:21]: nope, line 20 implies `len(parts) >= 3`

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**